app/services/property_search.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class PropertySearch
+  def initialize(params)
+    @id = params.fetch(:id, nil)
+  end
+
+  def perform
+    return id_query if id.present?
+  end
+
+private
+
+  attr_reader :id
+
+  def id_query
+    {
+      property: property,
+      cautionaryAlerts: cautionary_alerts
+    }
+  end
+
+  def cautionary_alerts
+    CautionaryAlert.for_property_reference(id).alerts
+  rescue Request::RecordNotFoundError
```

Comment:
As Neil suggested, would be great to add a comment here explaining why we are doing this